### ISSUE TITLE
using serve/watch the cache does not pick up changes in the parent template

### ISSUE THREAD

**Issue by webdiscus** [webdiscus/html-bundler-webpack-plugin#127]
# Description

The problem appears when we have many entry templates (multiple pages).

After a partial file is modified, the parent template will not be rebuild.
Sometimes the template will be rebuild  only after first change, but further changes make no effect.
Always  will be rebuild  only the template defined as last in the `entry` plugin option.

The issue is independent of cache type (filesystem or memory) and the preprocessor (template engine).

There are pages containing the same `header` partial:

```
page A > header
page B > header
```

### Scenario to reproduce the issue:

1st change `header` -> rebuild page A and B
2st change `header` -> rebuild  only last page B

# Problem

Webpack doesn't know which partials are used in which templates, so Webpack can't rebuild the main template (entrypoint) where a partial has changed. We do not have a dependence graph of partial on the main templates. Some template engines have a powerful API that allows you to walk through the dependency tree, such as Pug. Theoretically, it is possible to create a dependency graph for such template engines. But does anyone need this?

# Semi solution

We can check if changed file is a partial (has the file extension matched to `test` plugin option, depending on used template engine) and not a main template (entrypoint).

After a partial file is modified, **all entry point templates**, including templates that do not contain the modified partial file, will be rebuilt.

Yes, it is not optimal, but there is no other solution.

**Comment by webdiscus** [comment]
# We need your help!

Can you please test the feature/fix in your multiple page projects and give here feedback?

**Comment by Bessonov** [comment]
@webdiscus Thank you, I can confirm, that editing the parent handlebars template and the option
```js
cache: {
	type: 'filesystem',
},
```
work for me!

I am not sure that it is related, but sometimes it crashes with:
```
home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/html-bundler-webpack-plugin@4.10.1_handlebars@4.7.8_webpack@5.97.1/node_modules/html-bundler-webpack-plugin/src/Plugin/AssetCompiler.js:598
      const isFileWatchable = dependency.isFileWatchable(fileName);
                                         ^

TypeError: Cannot read properties of null (reading 'isFileWatchable')
    at HtmlBundlerPlugin.invalidate (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/html-bundler-webpack-plugin@4.10.1_handlebars@4.7.8_webpack@5.97.1/node_modules/html-bundler-webpack-plugin/src/Plugin/AssetCompiler.js:598:42)
    at Hook.eval [as call] (eval at create (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/tapable@2.2.1/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:9:1)
    at Watchpack.<anonymous> (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/webpack@5.97.1_webpack-cli@5.1.4/node_modules/webpack/lib/Watching.js:390:35)
    at Object.onceWrapper (node:events:633:26)
    at Watchpack.emit (node:events:518:28)
    at Watchpack._onChange (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/watchpack@2.4.2/node_modules/watchpack/lib/watchpack.js:364:9)
    at Watcher.<anonymous> (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/watchpack@2.4.2/node_modules/watchpack/lib/watchpack.js:89:15)
    at Watcher.emit (node:events:518:28)
    at /home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/watchpack@2.4.2/node_modules/watchpack/lib/DirectoryWatcher.js:220:41
    at DirectoryWatcher.forEachWatcher (/home/vagrant/workspace/website/.cache/pnpm/virtual-store-dir/watchpack@2.4.2/node_modules/watchpack/lib/DirectoryWatcher.js:126:5)

Node.js v22.11.0
```

However, I see there is a new version. I will try.

**Comment by webdiscus** [comment]
@Bessonov 

> ... sometimes it crashes ...

can you please provide a repo with reproducible issue and describe steps how to reproduce it?

**Comment by Bessonov** [comment]
@webdiscus 
> can you please provide a repo with reproducible issue and describe steps how to reproduce it?

Nope. After update no crashes anymore :+1:

**Comment by webdiscus** [comment]
@Bessonov the `Cannot read properties of null (reading 'isFileWatchable')` is fixed in the #132, `v4.10.4`
